assets/src/edit-story/components/library/panes/media/media3p/media3pCategories.js

```diff
@@ -130,9 +133,53 @@ const Media3pCategories = ({
       containerRef.current.style.height = '36px';
     } else {
       containerRef.current.style.height = `${innerContainerRef.current.offsetHeight}px`;
+      setFocusedRowOffset(0);
     }
   }, [containerRef, innerContainerRef, isExpanded]);
 
+  // Handles setting which row will be seen, by manipulating translateY.
```

Comment:
Would it make sense to extract the animation logic to its own hook fn that returns setFocusedRowOffset and focusedRowOffset to separate the animation logic from the rendering logic?

Reply:
This logic is not really reusable and is very intertwined with the representation of the category section and its pill, so I don't see a lot of value in spinning it to a separate file besides reducing the size of media3pPcategories.js

Reply:
If you feel strongly I can move them to a separate file though.

Reply:
I don't feel strongly about it, this was only an idea.